bundles/EcommerceFrameworkBundle/IndexService/Worker/ElasticSearch/AbstractElasticSearch.php

```diff
@@ -836,4 +840,126 @@ protected function createEsIndex(string $indexName)
         $this->settingsHash = md5(json_encode($this->tenantConfig->getIndexSettings()));
         $this->updateVersionFile();
     }
+
+    /**
+     * Get the next index version, e.g. if currently 13, then 14 will be returned.
+     * @return int
+     */
+    private function getNextIndexVersion() : int {
+        return $this->getIndexVersion()+1;
+    }
+
+    /**
+     * Build the name of the index for a specific index number.
+     * @param int $id e.g. 13
+     * @return string the name of the index, such as at_de_elastic_13
+     */
+    private function buildIndexName(int $id) : string {
```

Comment:
why not provide an optional parameter to `getIndexNameVersion` and use this method instead? 
```
    public function getIndexNameVersion($indexVersionOverride = null)
    {
        $indexVersion = $indexVersionOverride ?? $this->getIndexVersion()
        return $this->indexName . '-' . $indexVersion;
    }
```
snippet not tested though.

Reply:
God idea! So we got rid of the ``buildIndexName()`` method...